server/src/main/java/com/google/android/attestation/AuthorizationList.java

```diff
@@ -777,307 +783,117 @@ private static void addOptionalAttestationApplicationId(
   }
 
   public static Builder builder() {
-    return new Builder();
+    return new AutoValue_AuthorizationList.Builder()
+        .setPurpose(ImmutableSet.of())
+        .setDigest(ImmutableSet.of())
+        .setPadding(ImmutableSet.of())
+        .setRollbackResistance(false)
+        .setNoAuthRequired(false)
+        .setUserAuthType(ImmutableSet.of())
+        .setAllowWhileOnBody(false)
+        .setTrustedUserPresenceRequired(false)
+        .setTrustedConfirmationRequired(false)
+        .setUnlockedDeviceRequired(false)
+        .setAllApplications(false)
+        .setRollbackResistant(false)
+        .setIndividualAttestation(false)
+        .setIdentityCredentialKey(false)
+        .setUnorderedTags(ImmutableList.of());
   }
 
   /**
    * Builder for an AuthorizationList. Any field not set will be made an Optional.empty or set with
    * the default value.
    */
-  public static final class Builder {
-
-    ImmutableSet<OperationPurpose> purpose = ImmutableSet.of();
-    Algorithm algorithm;
-    Integer keySize;
-    ImmutableSet<DigestMode> digest = ImmutableSet.of();
-    ImmutableSet<PaddingMode> padding = ImmutableSet.of();
-    EcCurve ecCurve;
-    Long rsaPublicExponent;
-    boolean rollbackResistance;
-    Instant activeDateTime;
-    Instant originationExpireDateTime;
-    Instant usageExpireDateTime;
-    boolean noAuthRequired;
-    ImmutableSet<UserAuthType> userAuthType = ImmutableSet.of();
-    Duration authTimeout;
-    boolean allowWhileOnBody;
-    boolean trustedUserPresenceRequired;
-    boolean trustedConfirmationRequired;
-    boolean unlockedDeviceRequired;
-    boolean allApplications;
-    ByteString applicationId;
-    Instant creationDateTime;
-    KeyOrigin origin;
-    boolean rollbackResistant;
-    RootOfTrust rootOfTrust;
-    Integer osVersion;
-    Integer osPatchLevel;
-    AttestationApplicationId attestationApplicationId;
-    ByteString attestationApplicationIdBytes;
-    ByteString attestationIdBrand;
-    ByteString attestationIdDevice;
-    ByteString attestationIdProduct;
-    ByteString attestationIdSerial;
-    ByteString attestationIdImei;
-    ByteString attestationIdSecondImei;
-    ByteString attestationIdMeid;
-    ByteString attestationIdManufacturer;
-    ByteString attestationIdModel;
-    Integer vendorPatchLevel;
-    Integer bootPatchLevel;
-    boolean individualAttestation;
-    boolean identityCredentialKey;
-    ImmutableList<Integer> unorderedTags;
-
-    @CanIgnoreReturnValue
-    public Builder setPurpose(Set<OperationPurpose> purpose) {
-      this.purpose = ImmutableSet.copyOf(purpose);
-      return this;
-    }
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setPurpose(Set<OperationPurpose> purpose);
```

Comment:
Does this expand to do the same ImmutableSet.copyOf?

Reply:
go/autovalue/builders-howto#collection

Indeed it does.